### PR TITLE
Use non-privileged mode for compute

### PR DESCRIPTION
**Pull request by zack-shoylev**
Is this needed for the compute service adapter?

It's causing issues against one of the docker service providers I am testing with.

### CODE REVIEW

**Comment by nacx** [comment]
See comment: https://github.com/jclouds/jclouds-labs/pull/247#issuecomment-198543632

File: docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public DockerComputeServiceAdapter(DockerApi api) {
 
          HostConfig.Builder hostConfigBuilder = HostConfig.builder()
                  .publishAllPorts(true)
-                 .privileged(true);
+                 .privileged(false);
```
Comment (andreaturli) [inline docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java]: probably better to add a property in the DockerTemplateOptions

File: docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java [hunk 2]
```diff
@@ -94,9 +94,10 @@
    protected Map<Integer, Integer> portBindings = ImmutableMap.of();
    protected String networkMode;
    protected Map<String, String> extraHosts = ImmutableMap.of();
+   protected Boolean privileged;
```
Comment (nacx) [inline docker/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java]: Better make this a primitive boolean; there's no need to have to deal with nulls

**Comment by nacx** [comment]
Thanks @zack-shoylev! Just a comment; apart from that lgtm!

**Comment by zack-shoylev** [comment]
merged

**Pull request closed without merging**